File: app/src/api/scheduling/billing/data-transfer-objects/subscription-plan-price.ts
import { BillingInterval } from '@/api/scheduling/billing/data-transfer-objects/billing-interval';

/**
 * A price for a subscription plan.
 */
export class SubscriptionPlanPrice {
    constructor(public amount: number = 0, public interval: BillingInterval, public billingId: string) {}
}

File: api/src/Domain/Scheduling/Billing/ValueObjects/SubscriptionPlanPrice.cs
using System;

namespace DetailingArsenal.Domain.Scheduling.Billing {
    public class SubscriptionPlanPrice : ValueObject<SubscriptionPlanPrice> {
        public decimal Amount { get; }
        public string Interval { get; }
        public BillingReference BillingReference { get; }

        public SubscriptionPlanPrice(decimal amount, string interval, BillingReference billingReference) {
            Amount = amount;
            Interval = interval;
            BillingReference = billingReference;
        }
    }
}